### DISCUSSION TITLE
Microsoft App store distribution possible?

### DISCUSSION THREAD

**Discussion by OmarJay1** [MicrosoftEdge/WebView2Feedback#3865] [Q&A] [answered]
Is it possible to distribute an app that uses WebView2 in an msix App Store bundle?  

This page says you need to check for the dependency and install if necessary in your installation package.  I'm not sure how I would do that in an msix package using the Visual Studio Win32 msix packaging tool.

https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/distribution

Thanks.

Update: I found this link which I think will work.  

https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-desktop6-installactions

**Comment by OmarJay1** [comment]
Further update.  I believe that this sample might answer my question, although I'm still trying to implement it.  Thank you.

https://github.com/MicrosoftEdge/WebView2Samples/tree/master/SampleApps/WV2DeploymentVSInstallerSample

Further update again: As far as I can tell, none of the deployment samples listed in that repository deal with msix packaging that's necessary to create a store app.  Thanks.

**Comment by liminzhu** [comment]
Thanks for posting this issue @OmarJay1 . Store/MSIX support for WebView2 is lacking at the moment, as we're currently working on issues in the same category but with higher priority (shipping WebView2 inbox in future Windows versions, getting some plans together to best-effort push WV2 to in-market devices, etc. so that you have to deal with WebView2 Runtime install in less and less cases and one day you'll not have to worry about deploying runtime), but we will get there later.

One thing you can do right now, if you're a Win32 full trust Store app or MSIX app, is that if the runtime isn't already installed, you can download the installer from our CDN (or package the installer with your MSIX package) and then invoke it to install the runtime during app first launch. This requires elevation as WebView2 can only be installed per system right now.

**Comment by kinex** [comment]
@liminzhu Is it possible to publish UWP/C# app using WebView2 control in the Microsoft Store? I couldn't find a clear answer to this question anywhere.

**Comment by ysc3839** [comment]
IIRC the WebView2 in Windows 10X is disturbed using appx. But that version doesn't work on standard Windows 10.
I hope future versions of Windows 10 can also use appx and Microsoft Store for WebView2 updates.

**Comment by fosJoddie** [comment]
> Thanks for posting this issue @OmarJay1 . Store/MSIX support for WebView2 is lacking at the moment, as we're currently working on issues in the same category but with higher priority (shipping WebView2 inbox in future Windows versions, getting some plans together to best-effort push WV2 to in-market devices, etc. so that you have to deal with WebView2 Runtime install in less and less cases and one day you'll not have to worry about deploying runtime), but we will get there later.
> 
> One thing you can do right now, if you're a Win32 full trust Store app or MSIX app, is that if the runtime isn't already installed, you can download the installer from our CDN (or package the installer with your MSIX package) and then invoke it to install the runtime during app first launch. This requires elevation as WebView2 can only be installed per system right now.

@liminzhu 
This issue being open still, and Windows 11 being ready to ship soon; do you have any updates on this issue? 

Will Windows 11 in S mode be able to install a store distributed msix packaged App that utilizes Webbiew2? 

We have been holding off on migrating our WPF app to uwp because of the lack of webview2 support, and it seems that msix packaging could finally become a viable solution. 

Distribution on Windows in S mode is a big goal for us, and S mode will only run published store apps.

**Comment by thegrxp** [comment]
Any update?

**Comment by alexispolakMURAL** [comment]
Any updates on this? I'm using WinUI2 with WebView2 and the WebView2 Fixed Runtime and I'm getting similar messages when trying to publish a release build of a WebView2 app (the apps is just the webview2 and nothing more)

**Comment by KaiSoelter** [comment]
Windows Store refuses certification of on UWP App (msix) with WebView2 using latest
https://github.com/microsoft/microsoft-ui-xaml/releases/tag/vMicrosoft.Experimental.UI.Xaml.2.8.0-prerelease.220118001

![image](https://user-images.githubusercontent.com/80787809/154425321-4a966a03-b24f-4bc3-9bbc-7e355cd23a8b.png)
Not Good.

API _CxxThrowException in vcruntime140.dll is not supported for this application type. Microsoft.Web.WebView2.Core.dll calls this API.
API __CxxFrameHandler3 in vcruntime140.dll is not supported for this application type. Microsoft.Web.WebView2.Core.dll calls this API.
API __std_exception_copy in vcruntime140.dll is not supported for this application type. Microsoft.Web.WebView2.Core.dll calls this API.
....

Windows SDK is updated to latest SDK 10.0.22000.194. to no avail as of https://github.com/MicrosoftEdge/WebView2Feedback/issues/2160
and https://github.com/MicrosoftEdge/WebView2Feedback/issues/1868#issuecomment-953661664

**The app passes a locally run WACK.**

![image](https://user-images.githubusercontent.com/80787809/154834048-d8a6b07a-cfb4-4c1e-af26-4b2fa59ea3fb.png)
![image](https://user-images.githubusercontent.com/80787809/154834086-ab5869ea-afa0-4515-a8ce-437828b6be50.png)

Trying the RunFullTrust...mentioned in https://github.com/MicrosoftEdge/WebView2Feedback/issues/2160

**Comment by KaiSoelter** [comment]
Tried the runFullTrust. Crashes on Launch then
See. https://github.com/MicrosoftEdge/WebView2Feedback/issues/2160#issuecomment-1046805732

**Comment by kathdeas** [comment]
> @liminzhu Is it possible to publish UWP/C# app using WebView2 control in the Microsoft Store? I couldn't find a clear answer to this question anywhere.

Wondering if you have resolved this now? We are looking and not finding a solution :(

**Comment by PratyakshSinha** [comment]
Is there any update on this issue?

**Reply by Vict0r-Chen** [reply, marked as answer]
I think there is no limitation on publishing a UWP(C#) app based on WebView2  to the Windows Store, Because i published one and it was approved yesterday.😁
